src/tyr/route_serializer_osrm.cc

```diff
@@ -325,6 +334,47 @@ struct IntersectionEdges {
   }
 };
 
+// Forward declaration
+valhalla::baldr::json::MapPtr serialize_incident(const TripLeg_Edge_Incident& incident,
+                                                 bool only_id = false);
+
+// Serializes incidents and adds to json-document
+void addsIncidents(const EnhancedTripLeg_Edge* prev_edge,
+                   const EnhancedTripLeg_Edge& curr_edge,
+                   json::Jmap& doc) {
+  if (curr_edge.incidents().size() == 0) {
+    // No incidents, nothing to do
+    return;
+  }
+  // Remember incidents on previous edge, an incident that continues
+  // over many edges need only to display the id again
+  std::unordered_set<uint32_t> incidents_on_previous_edge;
+  if (prev_edge) {
+    for (const auto& prev_incident : prev_edge->incidents()) {
+      incidents_on_previous_edge.emplace(prev_incident.id());
+    }
+  }
+  for (const auto& incident : curr_edge.incidents()) {
+    json::ArrayPtr serialized_incidents = std::shared_ptr<json::Jarray>(new json::Jarray());
+    {
+      // Bring up any already existing array
+      auto existing = doc.find("incidents");
+      if (existing != doc.end()) {
+        if (auto* ptr =
+                boost::get<std::shared_ptr<valhalla::baldr::json::Jarray>>(&existing->second)) {
+          serialized_incidents = *ptr;
+        } else {
+          throw std::logic_error("This should not be possible");
+        }
+      }
+    }
+    bool only_id = incidents_on_previous_edge.count(incident.id()) > 0;
+    auto json_incident = serialize_incident(incident, only_id);
+    serialized_incidents->emplace_back(json_incident);
+    doc.emplace("incidents", serialized_incidents);
```

Comment:
Should this line not be outside the for loop?

Reply:
Yup, good catch, although it works anyway (although slightly ineffecient) due to fetching existing element from the json, as evidenced by the unittest. Pushing update.